File: charcutaria-business-manager/src/app/shared/services/order/order.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Order } from '../../models/order';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { OrderDetails } from '../../models/orderDetails';
import { UpdateOrder } from '../../models/updateOrder';
import { NewOrderItem } from '../../models/NewOrderItem';
import { OrderSummary } from '../../models/orderSummary';
import { orderSummaryFilter } from '../../models/orderSummaryFilter';
import { PagedResult } from '../../models/pagedResult';
import { OrderItemReportFilter } from '../../models/OrderItemReportFilter';
import { OrderItemReport } from '../../models/orderItemReport';
import { OrderCountSummary } from '../../models/orderCountSummary';
import { ProfitSummary } from '../../models/profitSummary';
import { SalesSummary } from '../../models/salesSummary';
import { PendingPaymentsSummary } from '../../models/pendingPaymentSummary';
import { SalesPerMonth } from '../../models/salesPerMonth';
import { SummarizedProductionFilter } from '../../models/summarizedProductionFilter';
import { SummarizedOrderReport } from '../../models/summarizedOrderReport';
import { PayOrder } from '../../models/payOrder';
import { RefundPayment } from '../../models/refundPayment';
import { map } from 'rxjs/operators';
import * as moment from 'moment-timezone';

@Injectable({
  providedIn: 'root'
})
export class OrderService {

  constructor(private httpClient: HttpClient) { }

  public create(order: Order): Observable<number> {
    return this.httpClient.post<number>(`${environment.apiUrl}/Order`, order);
  }
  public get(orderId: number): Observable<OrderDetails> {
    return this.httpClient.get<OrderDetails>(`${environment.apiUrl}/Order/${orderId}`).pipe(map(r => {
      var zone = moment.tz.guess();
      if (r.paidOn != null)
        r.paidOn = moment(r.paidOn).tz(zone).format();
      if (r.createdOn != null)
        r.createdOn = moment(r.createdOn).tz(zone).format();
      if (r.lastUpdated != null)
        r.lastUpdated = moment(r.lastUpdated).tz(zone).format();
      return r;
    }));
  }
  public getByNumber(orderNumber: number): Observable<OrderDetails> {
    return this.httpClient.get<OrderDetails>(`${environment.apiUrl}/Order/number/${orderNumber}`).pipe(map(r => {
      var zone = moment.tz.guess();
      if (r.paidOn != null)
        r.paidOn = moment(r.paidOn).tz(zone).format();
      if (r.createdOn != null)
        r.createdOn = moment(r.createdOn).tz(zone).format();
      if (r.lastUpdated != null)
        r.lastUpdated = moment(r.lastUpdated).tz(zone).format();
      return r;
    }));
  }
  public update(model: UpdateOrder): Observable<any> {
    return this.httpClient.put<OrderDetails>(`${environment.apiUrl}/Order`, model);
  }
  public addPayment(model: PayOrder): Observable<any> {
    return this.httpClient.post<OrderDetails>(`${environment.apiUrl}/Order/Payment`, model);
  }
  public refundPayment(model: RefundPayment): Observable<any> {
    return this.httpClient.post<OrderDetails>(`${environment.apiUrl}/Order/Refund`, model);
  }
  public cancel(orderNumber: number): Observable<any> {
    return this.httpClient.post<any>(`${environment.apiUrl}/Order/cancel/${orderNumber}`, {});
  }
  public restore(orderNumber: number): Observable<number> {
    return this.httpClient.post<number>(`${environment.apiUrl}/Order/restore/${orderNumber}`, {});
  }
  public close(orderNumber: number): Observable<number> {
    return this.httpClient.post<number>(`${environment.apiUrl}/Order/Close/${orderNumber}`, {});
  }
  public removeOrderItem(orderId: number, orderItemId: number): Observable<any> {
    return this.httpClient.delete(`${environment.apiUrl}/Order/${orderId}/Item/${orderItemId}`);
  }
  public addOrderItem(orderItem: NewOrderItem): Observable<number> {
    return this.httpClient.post<number>(`${environment.apiUrl}/Order/Item`, orderItem);
  }
  public updateOrderItem(orderItem: NewOrderItem): Observable<any> {
    return this.httpClient.put<any>(`${environment.apiUrl}/Order/Item`, orderItem);
  }
  public getOrderSummary(filter: orderSummaryFilter, page: number, pageSize: number): Observable<PagedResult<OrderSummary>> {
    let params = new HttpParams();
    if (filter.customer)
      params = params.append('customer', filter.customer);
    if (filter.createdOnFrom != null)
      params = params.append('createdOnFrom', filter.createdOnFrom);
    if (filter.createdOnTo != null)
      params = params.append('createdOnTo', filter.createdOnTo);
    if (filter.paidOnFrom != null)
      params = params.append('paidOnFrom', filter.paidOnFrom);
    if (filter.paidOnTo != null)
      params = params.append('paidOnTo', filter.paidOnTo);
    if (filter.completeByFrom != null)
      params = params.append('completeByFrom', filter.completeByFrom);
    if (filter.completeByTo != null)
      params = params.append('completeByTo', filter.completeByTo);
    (filter.paymentStatus ?? []).forEach(s => {
      params = params.append('paymentStatus', String(s.paymentStatusId));
    });
    (filter.orderStatus ?? []).forEach(s => {
      params = params.append('orderStatus', String(s.orderStatusId));
    });
    if (filter.orderBy != null)
      params = params.append('orderBy', String(filter.orderBy));
    if (filter.direction != null)
      params = params.append('direction', String(filter.direction));
    return this.httpClient.get<PagedResult<OrderSummary>>(`${environment.apiUrl}/Order/${page}/${pageSize}`, { params: params }).pipe(map(data => {
      var zone = moment.tz.guess();
      (data.data ?? []).forEach(r => {
        if (r.paidOn != null)
          r.paidOn = moment(r.paidOn).tz(zone).format();
        if (r.createdOn != null)
          r.createdOn = moment(r.createdOn).tz(zone).format();
      })
      return data;
    }));

  }
  public getOrderItemReport(filter: OrderItemReportFilter, page: number, pageSize: number): Observable<PagedResult<OrderItemReport>> {
    let params = new HttpParams();
    if (filter.orderNumber != null)
      params = params.append('orderNumber', String(filter.orderNumber));
    if (filter.customerId != null)
      params = params.append('customerId', String(filter.customerId));
    (filter.itemStatus ?? []).forEach(s => {
      params = params.append('itemStatus', String(s.orderItemStatusId));
    });
    (filter.products ?? []).forEach(s => {
      params = params.append('productId', String(s));
    });
    (filter.orderStatus ?? []).forEach(s => {
      params = params.append('orderStatus', String(s.orderStatusId));
    });
    if (filter.volumeUnitId != null)
      params = params.append('volumeUnitId', String(filter.volumeUnitId));
    if (filter.massUnitId != null)
      params = params.append('massUnitId', String(filter.massUnitId));
    if (filter.completeByFrom != null)
      params = params.append('completeByFrom', filter.completeByFrom);
    if (filter.completeByTo != null)
      params = params.append('completeByTo', filter.completeByTo);
    if (filter.customer)
      params = params.append('customer', filter.customer);
    if (filter.orderBy != null)
      params = params.append('orderBy', String(filter.orderBy));
    if (filter.direction != null)
      params = params.append('direction', String(filter.direction));
    return this.httpClient.get<PagedResult<OrderItemReport>>(`${environment.apiUrl}/Order/Report/Item/${page}/${pageSize}`, { params: params }).pipe(map(data => {
      var zone = moment.tz.guess();
      (data.data ?? []).forEach(r => {
        if (r.lastStatusDate != null)
          r.lastStatusDate = moment(r.lastStatusDate).tz(zone).format();
      });
      return data;
    }));

  }
  public getOrderCountSummary(): Observable<OrderCountSummary> {
    return this.httpClient.get<OrderCountSummary>(`${environment.apiUrl}/Order/Report/OrderCountSummary`);
  }
  public getProfitSummary(): Observable<ProfitSummary> {
    return this.httpClient.get<ProfitSummary>(`${environment.apiUrl}/Order/Report/ProfitSummary`);
  }
  public getSalesSummary(): Observable<SalesSummary> {
    return this.httpClient.get<SalesSummary>(`${environment.apiUrl}/Order/Report/SalesSummary`);
  }
  public getPendingPaymentsSummary(): Observable<PendingPaymentsSummary> {
    return this.httpClient.get<PendingPaymentsSummary>(`${environment.apiUrl}/Order/Report/PendingPaymentsSummary`);
  }

  public getSalesPerMonth(): Observable<SalesPerMonth[]> {
    return this.httpClient.get<SalesPerMonth[]>(`${environment.apiUrl}/Order/Report/SalesPerMonth`);
  }
  public getSummarizedReport(filter: SummarizedProductionFilter, page: number, pageSize: number): Observable<PagedResult<SummarizedOrderReport>> {
    let params = new HttpParams();
    (filter.itemStatus ?? []).forEach(s => {
      params = params.append('itemStatus', String(s.orderItemStatusId));
    });
    (filter.products ?? []).forEach(s => {
      params = params.append('products', String(s.productId));
    });
    if (filter.volumeUnitId != null)
      params = params.append('volumeUnitId', String(filter.volumeUnitId));
    if (filter.massUnitId != null)
      params = params.append('massUnitId', String(filter.massUnitId));
    if (filter.orderBy != null)
      params = params.append('orderBy', String(filter.orderBy));
    if (filter.direction != null)
      params = params.append('direction', String(filter.direction));
    return this.httpClient.get<PagedResult<SummarizedOrderReport>>(`${environment.apiUrl}/Order/Report/SummarizedProduction/${page}/${pageSize}`, { params: params });

  }
}

File: Charcutarie.Api/Charcutarie.Services/OrderService.cs
using Charcutarie.Application.Contracts;
using Charcutarie.Core.ExceptionHandling;
using Charcutarie.Models;
using Charcutarie.Models.Enums;
using Charcutarie.Models.Enums.OrderBy;
using Charcutarie.Models.ViewModels;
using Charcutarie.Services.Contracts;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.Transactions;

namespace Charcutarie.Services
{
    public class OrderService : IOrderService
    {
        private readonly IOrderApp orderApp;
        private readonly IPricingApp pricingApp;
        private readonly IProductApp productApp;
        private readonly IOrderItemApp orderItemApp;
        private readonly IMeasureUnitApp measureUnitApp;
        private readonly IDataSheetService dataSheetService;
        private readonly ITransactionApp transactionApp;

        public OrderService(IOrderApp orderApp,
            IPricingApp pricingApp,
            IProductApp productApp,
            IOrderItemApp orderItemApp,
            IMeasureUnitApp measureUnitApp,
            IDataSheetService dataSheetService,
            ITransactionApp transactionApp)
        {
            this.orderApp = orderApp;
            this.pricingApp = pricingApp;
            this.productApp = productApp;
            this.orderItemApp = orderItemApp;
            this.measureUnitApp = measureUnitApp;
            this.dataSheetService = dataSheetService;
            this.transactionApp = transactionApp;
        }
        public async Task<long> Add(NewOrder model, int corpClientId)
        {
            var ids = model.OrderItems.Select(i => i.ProductId).ToList();
            var prods = await productApp.GetRange(corpClientId, ids);
            int i = 1;
            var units = await measureUnitApp.GetAll();
            foreach (var item in model.OrderItems)
            {
                item.ItemNumber = i;
                var prod = prods.FirstOrDefault(p => p.ProductId == item.ProductId);
                item.ProductPrice = prod.Price;
                var pType = units.FirstOrDefault(u => u.MeasureUnitId == prod.MeasureUnitId).MeasureUnitTypeId;
                var qType = units.FirstOrDefault(u => u.MeasureUnitId == item.MeasureUnitId).MeasureUnitTypeId;
                await CalculateItem(item, prod.MeasureUnitId, pType, qType, corpClientId);
                i++;
            }
            return await orderApp.Add(model, corpClientId);
        }
        private async Task CalculateItem(NewOrderItem item, MeasureUnitEnum prodMeasureUnit, MeasureUnitTypeEnum pType, MeasureUnitTypeEnum qType, int corpClientId)
        {
            item.OriginalPrice = pricingApp.CalculatePricePerTotalWeight(new PriceRequest
            {
                ProductMeasureUnit = prodMeasureUnit,
                ProductPrice = item.ProductPrice,
                Quantity = item.Quantity,
                QuantityMeasureUnit = item.MeasureUnitId
            }, pType, qType);
            var prodSummary = await dataSheetService.CalculateProduction(item.ProductId, item.MeasureUnitId, item.Quantity, corpClientId);
            item.Cost = prodSummary.ProductionItems.Any() ? prodSummary.ProductionCost : new decimal?();
            item.Profit = prodSummary.ProductionItems.Any() ? (item.PriceAfterDiscount - prodSummary.ProductionCost) : new decimal?();
        }
        public async Task ChangeStatus(UpdateOrderStatus model, int corpClientId)
        {
            await orderApp.ChangeStatus(model, corpClientId);
            await orderItemApp.UpdateAllOrderItemStatus(model.OrderNumber, OrderItemStatusEnum.Cancelado, corpClientId);
        }

        public async Task<Order> Get(long orderId, int corpClientId)
        {
            return await orderApp.Get(orderId, corpClientId);
        }

        public async Task<Order> GetByNumber(long orderNumber, int corpClientId)
        {
            return await orderApp.GetByNumber(orderNumber, corpClientId);
        }
        public async Task AddPayment(PayOrder model, int corpClientId, long userId)
        {
            if (model.Amount <= 0)
                throw new BusinessException("O valor do pagamento deve ser maior que zero");

            var currentOrder = await GetByNumber(model.OrderNumber, corpClientId);
            var payments = await transactionApp.GetTransactions(currentOrder.OrderId, corpClientId);
            var totalPaid = payments.Where(p => (p.IsOrderPrincipalAmount ?? false)).Sum(s => s.Amount);
            if (totalPaid + model.Amount > currentOrder.ItemsTotalAfterDiscounts)
                throw new BusinessException("O valor total pago não pode ser maior que o total da venda");

            var nextPaymentStatus = (totalPaid + model.Amount == currentOrder.ItemsTotalAfterDiscounts) ?
                PaymentStatusEnum.Pago : PaymentStatusEnum.ParcialmentePago;

            using var trans = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
            await orderApp.Update(new UpdateOrder
            {
                CompleteBy = currentOrder.CompleteBy,
                FreightPrice = currentOrder.FreightPrice,
                OrderNumber = currentOrder.OrderNumber,
                PaymentStatusId = nextPaymentStatus
            }, corpClientId);

            await transactionApp.AddTransaction(new NewTransaction
            {
                Amount = model.Amount,
                CorpClientId = corpClientId,
                Date = DateTime.Now,
                Description = $"Recebimento referente ao Pedido #{currentOrder.OrderNumber}",
                MerchantName = currentOrder.Customer.Name,
                OrderId = currentOrder.OrderId,
                TransactionTypeId = model.OrderPaymentMethod,
                TransactionStatusId = 1,
                IsOrderPrincipalAmount = true
            }, userId);
            if (model.Tip > 0)
            {
                await transactionApp.AddTransaction(new NewTransaction
                {
                    Amount = model.Tip,
                    CorpClientId = corpClientId,
                    Date = DateTime.Now,
                    Description = $"Recebimento de Tips referente ao Pedido #{currentOrder.OrderNumber}",
                    MerchantName = currentOrder.Customer.Name,
                    OrderId = currentOrder.OrderId,
                    TransactionTypeId = model.TipPaymentMethod,
                    TransactionStatusId = 1,
                    IsOrderPrincipalAmount = false
                }, userId);
            }
            trans.Complete();
        }
        public async Task RefundPayment(RefundPayment model, int corpClientId, long userId)
        {

            var currentOrder = await GetByNumber(model.OrderNumber, corpClientId);
            var payments = await transactionApp.GetTransactions(currentOrder.OrderId, corpClientId);
            var transaction = payments.FirstOrDefault(p => p.TransactionId == model.TransactionId);

            if (transaction.TransactionStatusId != 1)
                throw new BusinessException("Não é possível fazer estorno desta transação.");

            var totalPaid = payments.Where(s => s.IsIncome && s.TransactionStatusId == 1).Sum(s => s.Amount);

            var nextPaymentStatus = (PaymentStatusEnum)currentOrder.PaymentStatusId;
            if ((transaction.IsOrderPrincipalAmount ?? false))
            {
                if ((currentOrder.ItemsTotalAfterDiscounts - transaction.Amount) == 0)
                    nextPaymentStatus = PaymentStatusEnum.Estornado;
                else if ((currentOrder.ItemsTotalAfterDiscounts - transaction.Amount) == currentOrder.ItemsTotalAfterDiscounts)
                    nextPaymentStatus = PaymentStatusEnum.Pago;
                else if (totalPaid - transaction.Amount <= 0)
                    nextPaymentStatus = PaymentStatusEnum.Pendente;
                else
                    nextPaymentStatus = PaymentStatusEnum.ParcialmentePago;
            }

            using var trans = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
            await orderApp.Update(new UpdateOrder
            {
                CompleteBy = currentOrder.CompleteBy,
                FreightPrice = currentOrder.FreightPrice,
                OrderNumber = currentOrder.OrderNumber,
                PaymentStatusId = nextPaymentStatus
            }, corpClientId);
            await transactionApp.ChangeStatus(transaction.TransactionId, corpClientId, 2);
            await transactionApp.AddTransaction(new NewTransaction
            {
                Amount = transaction.Amount,
                CorpClientId = corpClientId,
                Date = DateTime.Now,
                Description = $"Transação (#{transaction.TransactionId}) do pedido #{currentOrder.OrderNumber} estornada",
                MerchantName = transaction.MerchantName,
                OrderId = transaction.OrderId,
                TransactionTypeId = 9,
                TransactionStatusId = 1,
                IsOrderPrincipalAmount = transaction.IsOrderPrincipalAmount
            }, userId);
            trans.Complete();
        }

        public async Task Update(UpdateOrder model, int corpClientId)
        {
            await orderApp.Update(model, corpClientId);
        }
        public async Task<OrderStatusEnum> RestoreOrderStatus(long orderNumber, int corpClientId)
        {
            var status = await orderApp.GetCurrentStatus(orderNumber, corpClientId);
            if (status != OrderStatusEnum.Cancelado) return status;
            var nextStatus = await GetNextOrderStatus(orderNumber, corpClientId);
            await orderApp.ChangeStatus(new UpdateOrderStatus
            {
                OrderNumber = orderNumber,
                OrderStatusId = nextStatus
            }, corpClientId);
            return nextStatus;
        }
        private async Task<OrderStatusEnum> GetNextOrderStatus(long orderNumber, int corpClientId)
        {
            var nextStatus = OrderStatusEnum.Criado;
            var orderItems = await orderItemApp.GetAll(orderNumber, corpClientId);
            if (!orderItems.Any()) return nextStatus;
            if (orderItems.All(i => i.OrderItemStatusId == OrderItemStatusEnum.AguardandoProducao))
                nextStatus = OrderStatusEnum.Criado;
            else if (orderItems.All(i => i.OrderItemStatusId == OrderItemStatusEnum.Entregue || i.OrderItemStatusId == OrderItemStatusEnum.Cancelado))
                nextStatus = OrderStatusEnum.Finalizado;
            else if (orderItems.All(i => i.OrderItemStatusId == OrderItemStatusEnum.ProntoParaEntrega || i.OrderItemStatusId == OrderItemStatusEnum.Cancelado || i.OrderItemStatusId == OrderItemStatusEnum.Entregue))
                nextStatus = OrderStatusEnum.AguardandoEntrega;
            else if (orderItems.Any(i => i.OrderItemStatusId == OrderItemStatusEnum.EmAndamento
            || i.OrderItemStatusId == OrderItemStatusEnum.ProntoParaEntrega
            || i.OrderItemStatusId == OrderItemStatusEnum.Entregue))
                nextStatus = OrderStatusEnum.EmAndamento;
            return nextStatus;
        }

        public async Task RemoveOrderItem(long orderId, long orderItemId, int corpClientId)
        {
            await orderItemApp.Remove(orderId, orderItemId, corpClientId);
            var order = await orderApp.Get(orderId, corpClientId);
            var nextStatus = await GetNextOrderStatus(order.OrderNumber, corpClientId);
            await orderApp.ChangeStatus(new UpdateOrderStatus
            {
                OrderNumber = order.OrderNumber,
                OrderStatusId = nextStatus
            }, corpClientId);
        }

        public async Task UpdateOrderItem(UpdateOrderItem model, int corpClientId)
        {
            var item = await orderItemApp.Get(model.OrderItemId, corpClientId);
            var prod = await productApp.Get(corpClientId, model.ProductId);
            model.ProductPrice = item.ProductPrice;
            var units = await measureUnitApp.GetAll();
            var pType = units.FirstOrDefault(u => u.MeasureUnitId == prod.MeasureUnitId).MeasureUnitTypeId;
            var qType = units.FirstOrDefault(u => u.MeasureUnitId == model.MeasureUnitId).MeasureUnitTypeId;
            if (model.OrderItemStatusId == OrderItemStatusEnum.Cancelado)
            {
                model.Cost = null;
                model.Profit = null;
            }
            else
                await CalculateItem(model, prod.MeasureUnitId, pType, qType, corpClientId);

            await orderItemApp.Update(model, corpClientId);
            var nextStatus = await GetNextOrderStatus(item.Order.OrderNumber, corpClientId);
            await orderApp.ChangeStatus(new UpdateOrderStatus
            {
                OrderNumber = item.Order.OrderNumber,
                OrderStatusId = nextStatus
            }, corpClientId);
        }

        public async Task<long> AddOrderItem(NewOrderItem model, int corpClientId)
        {
            var order = await orderApp.Get(model.OrderId, corpClientId);
            var prod = await productApp.Get(corpClientId, model.ProductId);
            model.ProductPrice = prod.Price;
            model.ItemNumber = await orderItemApp.GetLastItemNumber(order.OrderNumber, corpClientId) + 1;
            var units = await measureUnitApp.GetAll();
            var pType = units.FirstOrDefault(u => u.MeasureUnitId == prod.MeasureUnitId).MeasureUnitTypeId;
            var qType = units.FirstOrDefault(u => u.MeasureUnitId == model.MeasureUnitId).MeasureUnitTypeId;
            await CalculateItem(model, prod.MeasureUnitId, pType, qType, corpClientId);
            var result = await orderItemApp.AddOrderItem(model, corpClientId);
            var nextStatus = await GetNextOrderStatus(order.OrderNumber, corpClientId);
            await orderApp.ChangeStatus(new UpdateOrderStatus
            {
                OrderNumber = order.OrderNumber,
                OrderStatusId = nextStatus
            }, corpClientId);
            return result;
        }

        public PagedResult<OrderSummary> GetOrderSummary(int corpClientId, string customer, DateTime? createdOnFrom, DateTime? createdOnTo, DateTime? paidOnFrom, DateTime? paidOnTo, DateTime? completeByFrom, DateTime? completeByTo, List<int> paymentStatus, List<int> orderStatus, OrderSummaryOrderBy orderBy, OrderByDirection direction, int? page, int? pageSize)
        {
            return orderApp.GetOrderSummary(corpClientId, customer, createdOnFrom, createdOnTo, paidOnFrom, paidOnTo, completeByFrom, completeByTo, paymentStatus, orderStatus, orderBy, direction, page, pageSize);
        }

        public PagedResult<OrderItemReport> GetOrderItemReport(int corpClientId, int? orderNumber, List<long> productIds, int massUnitId, int volumeUnitId, List<OrderStatusEnum> orderStatus, List<OrderItemStatusEnum> itemStatus, DateTime? completeByFrom, DateTime? completeByTo, string customer, long? customerId, OrderItemReportOrderBy orderBy, OrderByDirection direction, int? page, int? pageSize)
        {
            return orderApp.GetOrderItemReport(corpClientId, orderNumber, productIds, massUnitId, volumeUnitId, orderStatus, itemStatus, completeByFrom, completeByTo, customer, customerId, orderBy, direction, page, pageSize);
        }
        public async Task CloseOrder(long orderNumber, int corpClientId)
        {
            await orderItemApp.UpdateAllOrderItemStatus(orderNumber, OrderItemStatusEnum.Entregue, corpClientId);
            var nextStatus = await GetNextOrderStatus(orderNumber, corpClientId);
            await orderApp.ChangeStatus(new UpdateOrderStatus
            {
                OrderNumber = orderNumber,
                OrderStatusId = nextStatus
            }, corpClientId);
        }

        public async Task<OrderCountSummary> GetOrderCountSummary(int corpClientId)
        {
            return await orderApp.GetOrderCountSummary(corpClientId);
        }

        public async Task<ProfitSummary> GetProfitSummary(int corpClientId)
        {
            return await orderApp.GetProfitSummary(corpClientId);
        }

        public async Task<SalesSummary> GetSalesSummary(int corpClientId)
        {
            return await orderApp.GetSalesSummary(corpClientId);
        }

        public async Task<PendingPaymentsSummary> GetPendingPaymentsSummary(int corpClientId)
        {
            return await orderApp.GetPendingPaymentsSummary(corpClientId);
        }

        public async Task<IEnumerable<SalesPerMonth>> GetSalesPerMonth(int corpClientId)
        {
            return await orderApp.GetSalesPerMonth(corpClientId);
        }
        public PagedResult<SummarizedOrderReport> GetSummarizedReport(int corpClientId, int volumeUnitId, int massUnitId, List<OrderItemStatusEnum> itemStatus, List<long> productIds, SummarizedOrderOrderBy orderBy, OrderByDirection direction, int? page, int? pageSize)
        {
            return orderApp.GetSummarizedReport(corpClientId, volumeUnitId, massUnitId, itemStatus, productIds, orderBy, direction, page, pageSize);
        }
    }
}
